testing/test_collection.py

```diff
@@ -49,31 +63,33 @@ def test_fail(): assert 0
             assert [1, 2, 3] != fn  # type: ignore[comparison-overlap]
             assert modcol != fn
 
-        assert testdir.collect_by_name(modcol, "doesnotexist") is None
+        assert pytester.collect_by_name(modcol, "doesnotexist") is None
 
-    def test_getparent(self, testdir):
-        modcol = testdir.getmodulecol(
+    def test_getparent(self, pytester: Pytester) -> None:
+        modcol = pytester.getmodulecol(
             """
             class TestClass:
                  def test_foo(self):
                      pass
         """
         )
-        cls = testdir.collect_by_name(modcol, "TestClass")
-        fn = testdir.collect_by_name(testdir.collect_by_name(cls, "()"), "test_foo")
+        cls = pytester.collect_by_name(modcol, "TestClass")
+        fn = pytester.collect_by_name(
+            pytester.collect_by_name(cls, "()"), "test_foo"  # type: ignore[arg-type]
+        )
 
-        parent = fn.getparent(pytest.Module)
+        parent = fn.getparent(pytest.Module)  # type: ignore[union-attr]
```

Comment:
I think we can avoid the ignores here if we add more asserts and avoiding reusing the `parent` variable name for values of different types. I suggest this:

```py
        cls = pytester.collect_by_name(modcol, "TestClass")
        assert isinstance(cls, pytest.Class)
        instance = pytester.collect_by_name(cls, "()")
        assert isinstance(instance, pytest.Instance)
        fn = pytester.collect_by_name(instance, "test_foo")
        assert isinstance(fn, pytest.Function)

        module_parent = fn.getparent(pytest.Module)
        assert module_parent is modcol

        function_parent = fn.getparent(pytest.Function)
        assert function_parent is fn

        class_parent = fn.getparent(pytest.Class)
        assert class_parent is cls
```

Reply:
I have made this change.